server/app/repository/ApplicationEventRepository.java

```diff
@@ -68,6 +64,58 @@ public ImmutableList<ApplicationEventModel> getEventsOrderByCreateTimeDesc(Long
             .findList());
   }
 
+  /**
+   * Updates the ApplicationEvents and the Applications table to the latest status in a single
+   * transaction.
+   *
+   * <p>Previously in 44.sql, DB triggers were used to update the latest_status between
+   * application_events table and applications table. In 78.sql script, the team decided to drop
+   * triggers and use application code to update both tables. This method implements the same.
+   *
+   * <p>Note - Application code must change both tables at once as we want to avoid inconsistency
+   * between the tables.
+   */
+  public CompletionStage<ApplicationEventModel> insertStatusEvent(
```

Comment:
(question) This look to only ever be used by forcing it to resolve the future by calling ` toCompletableFuture().join()`. Is these a reason to have this return a `CompletionStage<ApplicationEventModel>` instead of just `ApplicationEventModel`?

Reply:
The ApplicantService uses the asynchronously, since it used to use the original `supplyAsync()` method that this PR removes.